glean-core/python/glean/glean.py

```diff
@@ -132,56 +132,60 @@ def initialize(
 
         cls._upload_enabled = upload_enabled
 
-        cfg = _ffi.make_config(
-            cls._data_dir,
-            application_id,
-            cls._upload_enabled,
-            configuration.max_events,
-        )
+        # Use `Glean._execute_task` rather than `Glean.launch` here, since we
+        # never want to put this work on the `Dispatcher._preinit_queue`.
+        @Dispatcher._execute_task
+        def initialize():
+            cfg = _ffi.make_config(
+                cls._data_dir,
+                application_id,
+                cls._upload_enabled,
+                configuration.max_events,
+            )
+
+            cls._initialized = _ffi.lib.glean_initialize(cfg) != 0
+
+            # If initialization of Glean fails, we bail out and don't initialize
+            # further
+            if not cls._initialized:
+                return
+
+            for ping in cls._ping_type_queue:
```

Comment:
Is this threadsafe in Python? I'm worried about hitting the same trouble we were hitting in Kotlin. If it's thread-safe, can we add a comment about it and mentioned that we don't need to worry as we do for Kotlin?

Reply:
With the GIL, pretty much everything is threadsafe -- that doesn't eliminate race conditions, but there's no data structures that aren't threadsafe.  Given my earlier comment, though, I wonder if we need this at all...

Reply:
Will you investigate this in a follow-up?

Reply:
https://bugzilla.mozilla.org/show_bug.cgi?id=1641558